sequencing/ATACSeq/jobSubmission/6_batchRunGenotypeCheck.sh

```diff
@@ -0,0 +1,157 @@
+## - For modules or references required, please refer to each subscript run in this script.                                    ||
+## - Subscripts to be in ${SUB_SCRIPTS_DIR} = ./subscripts                                                                     ||
+## - Subscripts: compareBamWithGenotypes.sh, searchBestGenoMatch.sh                                                            ||
+## - R subscripts to be in ${RSCRIPTS_DIR} = ./Rscripts                                                                        ||
+## - R scripts: collateSampleChecks.Rmd                                                                                        ||
+##                                                                                                                             ||
+## ============================================================================================================================##
+
+## ============ ##
+##    SET UP    ##
+## ============ ##
+
+## print start date and time
+echo Job started on:
+date -u
+
+## load config file provided on command line related to the specified project
+source "${1}/config.txt"
+echo "Loading config file for project: " ${PROJECT}
+echo "Project directory is: " $MAIN_DIR
+echo 'Script is running from directory: ' ${SCRIPTS_DIR}
+
+LOG_DIR=${LOG_DIR}/${USER}/${SLURM_ARRAY_JOB_ID}
+echo "Log files will be moved to dir: " $LOG_DIR
+mkdir -p $LOG_DIR
+mv ATACGenoConcorS6-${SLURM_ARRAY_JOB_ID}_${SLURM_ARRAY_TASK_ID}* $LOG_DIR
+
+## ================ ##
+##    VARIABLES     ##
+## ================ ##
+
+if [[ $2 == '' ]] 
+then
+  echo "No step specified. All steps in the script will be run. "
+fi
+
+##check array specified and exit if not
+if [[ ${SLURM_ARRAY_TASK_ID} == '' ]]
+then 
+    { echo "Job does not appear to be an array. Please specify --array on the command line." ; exit 1; }
+fi
+
+## check step method matches required options and exit if not
+if [[ ! $2 =~ "COMPARE" ]] && [[ ! $2 =~ "GENCHECK" ]] && [[ ! $2 =~ "SWITCH" ]] &&[[ ! $2 == '' ]];
+then 
+    { echo "Unknown step specified. Please use COMPARE, GENCHECK, SWITCH or some combination of this as a single string (i.e. SPLIT,CHECK)" ; exit 1; }            
+fi
+
+## ============ ##
+##    STEPS     ##
+## ============ ##
+
+## option COMPARE: genotype of samples are verified by comparing them with existing genotype information   
+if [ $# = 1 ] || [[ $2 =~ 'COMPARE' ]]
+then
+  module purge
+  module load $PICARDVERS
+  module load GATK
+  module load $SAMTVERS
+  
+  echo " "
+  echo "|| Running STEP 6.1 of ATAC-seq pipeline: COMPARE. Samples will be compared to their matching genotype data.||"
+  echo " "
+  echo "Output directory will be: ${ALIGNED_DIR}/genotypeConcordance"
+  echo "Output directory will be: ${ALIGNED_DIR}/baseRecalibrate"
+  echo " "
+  
+  # process a line from IDMap file
+  IDS=($(head -n ${SLURM_ARRAY_TASK_ID} ${META_DIR}/matchedVCFIDs.txt | tail -1))
```

Comment:
Just a quick check, can `${SLURM_ARRAY_TASK_ID}` here be 1? If it can be then the first row of `matchedVCFIDs.txt` is a header row and so this won't give you what you want. 

Looking at an example of `samples.txt` the file doesn't have a header row, so I presume this will be pulling out the incorrect sample for all task ids (provided that my assumption that  `${SLURM_ARRAY_TASK_ID}` corresponds to the sample in `samples.txt` with the same row number).

I might be being silly and this works regardless due to 0/1 indexing, just checking.

Reply:
I guess it will make more sense to add a header row to samples.txt so that the first sample will be index 1 and not 0. My programming side assumes that the first index is 0 XD

Reply:
You can do that, sure. But then `${SLURM_ARRAY_TASK_ID}` will need to be `2` to select the first data row in `samples.txt` and `matchedVCFIDs.txt` (as `head -1 $file | tail -1` gets you the first row which will now be the header).